### PR TITLE
bpo-35631: Improve typing docs wrt abstract/concrete collection types

### PR DESCRIPTION
**Pull request by scop**
https://bugs.python.org/issue35631

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -678,10 +684,8 @@ The module defines the following classes, functions and decorators:
 .. class:: Dict(dict, MutableMapping[KT, VT])
 
    A generic version of :class:`dict`.
-   The usage of this type is as follows::
-
-      def get_position_in_index(word_list: Dict[str, int], word: str) -> int:
-          return word_list[word]
+   Useful for annotating return types. To annotate arguments it is preferred
+   to use an abstract collection type such as :class:`Mapping`.
```
Comment (ilevkivskyi) [inline Doc/library/typing.rst:688]: I would keep an example here, but update it to use a return annotation:
```python
def count_words(text: str) -> Dict[str, int]:
    ...
```

**Review by ilevkivskyi — COMMENTED** [review]
Thanks for the PR! I have just one little suggestion.

File: Doc/library/typing.rst (hunk 1 above)
Comment (scop) [reply Doc/library/typing.rst:688]: Done.

**Review by ilevkivskyi — APPROVED** [review]
Thanks for update, LGTM!

**Pull request merged**